airflow/jobs/scheduler_job.py

```diff
@@ -631,16 +632,16 @@ def create_dag_run(self, dag, session=None):
                 return next_run
 
     @provide_session
-    def _process_task_instances(self, dag, task_instances_list, session=None):
+    def _process_task_instances(self, dag, dag_runs, session=None):
```

Comment:
Can we add type annotations here as well?

Reply:
In this PR, I want to focus only on necessary changes. I want to add type-hint in separate PR, because type-hint in the core are problematic. They cause cyclical imports.